src/browser/new-browser.ts

```diff
@@ -1,51 +1,53 @@
 import { URLSearchParams } from "url";
 
 import URI from "urijs";
-import { isBoolean, assign, isEmpty } from "lodash";
+import { isBoolean, assign, isEmpty, set } from "lodash";
 import { remote, RemoteOptions } from "webdriverio";
 
 import { Browser, BrowserOpts } from "./browser";
 import signalHandler from "../signal-handler";
 import { runGroup } from "./history";
 import { warn } from "../utils/logger";
+import { getNormalizedBrowserName } from "../utils/browser";
 import { getInstance } from "../config/runtime-config";
-import { DEVTOOLS_PROTOCOL, WEBDRIVER_PROTOCOL, LOCAL_GRID_URL } from "../constants/config";
+import {
+    DEVTOOLS_PROTOCOL,
+    WEBDRIVER_PROTOCOL,
+    LOCAL_GRID_URL,
+    W3C_CAPABILITIES,
+    VENDOR_CAPABILITIES,
+} from "../constants/config";
 import { Config } from "../config";
 import { BrowserConfig } from "../config/browser-config";
 import { gridUrl as DEFAULT_GRID_URL } from "../config/defaults";
-
-export type CapabilityName = "goog:chromeOptions" | "moz:firefoxOptions" | "ms:edgeOptions";
-export type HeadlessBrowserOptions = Record<
-    string,
-    {
-        capabilityName: CapabilityName;
-        getArgs: (headlessMode: BrowserConfig["headless"]) => string[];
-    }
+import { BrowserName, type W3CBrowserName } from "./types";
+
+export type VendorSpecificCapabilityName = "goog:chromeOptions" | "moz:firefoxOptions" | "ms:edgeOptions";
+export type HeadlessBrowserOptions = Partial<
+    Record<
+        W3CBrowserName,
+        {
+            capabilityName: VendorSpecificCapabilityName;
+            getArgs: (headlessMode: BrowserConfig["headless"]) => string[];
+        }
+    >
 >;
 const DEFAULT_PORT = 4444;
 
 const headlessBrowserOptions: HeadlessBrowserOptions = {
-    chrome: {
+    [BrowserName.CHROME]: {
         capabilityName: "goog:chromeOptions",
         getArgs: (headlessMode: BrowserConfig["headless"]): string[] => {
             const headlessValue = isBoolean(headlessMode) ? "headless" : `headless=${headlessMode}`;
 
             return [headlessValue, "disable-gpu"];
         },
     },
-    firefox: {
+    [BrowserName.FIREFOX]: {
         capabilityName: "moz:firefoxOptions",
         getArgs: (): string[] => ["-headless"],
     },
-    msedge: {
-        capabilityName: "ms:edgeOptions",
-        getArgs: (): string[] => ["--headless"],
-    },
-    edge: {
-        capabilityName: "ms:edgeOptions",
-        getArgs: (): string[] => ["--headless"],
-    },
-    microsoftedge: {
+    [BrowserName.EDGE]: {
         capabilityName: "ms:edgeOptions",
         getArgs: (): string[] => ["--headless"],
     },
```

Comment:
Refactored "headlessBrowserOptions" so its key is valid w3c browserName, which is received by "getNormalizedBrowserName" function, which is moved from "src/browser-installer" to "src/utils/browser"